src/Features/CSharp/Portable/Completion/KeywordRecommenders/AssemblyKeywordRecommender.cs

```diff
@@ -17,15 +15,21 @@ public AssemblyKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            if (context.IsTypeAttributeContext(cancellationToken))
+            var token = context.TargetToken;
+
+            if (token.Kind() != SyntaxKind.OpenBracketToken)
             {
-                var token = context.LeftToken;
-                var type = token.GetAncestor<MemberDeclarationSyntax>();
+                return false;
+            }
 
-                return type == null || type.IsParentKind(SyntaxKind.CompilationUnit);
+            if (token.Parent.Kind() != SyntaxKind.AttributeList)
+            {
+                return true;
             }
 
-            return false;
+            var attributeList = token.Parent;
+            var previousSyntax = attributeList.Parent;
+            return previousSyntax is CompilationUnitSyntax || previousSyntax.Parent is CompilationUnitSyntax;
```

Comment:
i don't understand why we need to check both attributeList.Parent and attributeList.Parent.Parent.  What are the two cases for?

Reply:
There are some cases where `previousSyntax`  is Class/Struct/Interface/Enum Declaration for this case the parent  `previousSyntax` of is `CompilationUnitSyntax`. Is it clear? Or Would you like I try to explain better? Should I rename these variable? For something like declarationSyntax?

Reply:
Basically, the code itself is not clear as to why it works like it does.  It either needs comments explaining things, or it needs to be reorganized to be clearer as to what it's going.